api/src/main/java/org/openmrs/obs/handler/AbstractHandler.java

```diff
@@ -146,12 +148,21 @@ public boolean purgeComplexData(Obs obs) {
 	 * @param obs
 	 * @return File object
 	 */
+
 	public static File getComplexDataFile(Obs obs) {
+		if(obs.getValueComplex()!=null && !obs.getValueComplex().isEmpty()) {
```

Comment:
@dkayiwa  Introduced a line here just to make sure that the  file name is never empty even after removing the file extension